trace/opencensus/opencensus.go

```diff
@@ -46,7 +47,30 @@ func (s *span) End() {
 	s.s.End()
 }
 
-func (s *span) SetStatus(status trace.Status) {
+func (s *span) SetStatus(err error) {
+	if !s.s.IsRecordingEvents() {
+		return
+	}
+
+	var status octrace.Status
+
+	if err == nil {
+		status.Code = octrace.StatusCodeOK
+		s.s.SetStatus(status)
+		return
+	}
+
+	switch {
+	case errdefs.IsNotFound(err):
+		status.Code = octrace.StatusCodeNotFound
```

Comment:
Does it make more sense to have an (anonymous) interface a la:

```
type ocstatusError interface {
    OCStatus() octrace.Status
}
```
And then errdefs can just implement it?

Reply:
This ties errors to the opencensus implementation which is not great.
Since we already know the kind of error and can do the translation, this seems a bit heavy to add.